### ISSUE TITLE
ApiException: Could not retrieve the response body from the returned JSON

### ISSUE THREAD

**Issue by atomicalsoftwares** [aws-amplify/amplify-android#377]
Greetings,

When my computer is connected to the network, but doesn't have accessibility to the internet, the following Amplify error crashes the application.

https://pastebin.com/6w3vPjQg
```
04-11 15:50:58.539 6579-6608/ai.deepseed.sandbox E/AndroidRuntime: FATAL EXCEPTION: pool-3-thread-2
    Process: ai.deepseed.sandbox, PID: 6579
    java.lang.RuntimeException: com.amplifyframework.datastore.DataStoreException: Failure performing sync query to AppSync.
        at io.reactivex.internal.util.ExceptionHelper.wrapOrThrow(ExceptionHelper.java:46)
        at io.reactivex.internal.observers.BlockingMultiObserver.blockingGet(BlockingMultiObserver.java:93)
        at io.reactivex.Completable.blockingAwait(Completable.java:1227)
        at com.amplifyframework.datastore.syncengine.Orchestrator.lambda$start$0$Orchestrator(Orchestrator.java:3)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$Orchestrator$sXOpGcCGs9ZiNIMKtTT6tZMPTQU.run(lambda)
        at io.reactivex.internal.operators.completable.CompletableFromAction.subscribeActual(CompletableFromAction.java:35)
        at io.reactivex.Completable.subscribe(Completable.java:2309)
        at io.reactivex.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
        at io.reactivex.internal.operators.completable.CompletableCreate$Emitter.onComplete(CompletableCreate.java:64)
        at com.amplifyframework.datastore.AWSDataStorePlugin.lambda$null$2(AWSDataStorePlugin.java:1)
        at com.amplifyframework.datastore.-$$Lambda$AWSDataStorePlugin$KzWoRbIa5fAf-opMJmvpcYTLpM4.accept(lambda)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$null$0$SQLiteStorageAdapter(SQLiteStorageAdapter.java:3)
        at com.amplifyframework.datastore.storage.sqlite.-$$Lambda$SQLiteStorageAdapter$w4lkRK7kn70UKyGuaWOFqFrETHI.run(lambda)
        at io.reactivex.internal.observers.CallbackCompletableObserver.onComplete(CallbackCompletableObserver.java:53)
        at io.reactivex.internal.operators.completable.CompletableFromSingle$CompletableFromSingleObserver.onSuccess(CompletableFromSingle.java:51)
        at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback$FlatMapSingleObserver.onSuccess(SingleFlatMap.java:111)
        at io.reactivex.internal.operators.single.SingleCreate$Emitter.onSuccess(SingleCreate.java:67)
        at com.amplifyframework.datastore.storage.sqlite.PersistentModelVersion.lambda$null$1(PersistentModelVersion.java:1)
        at com.amplifyframework.datastore.storage.sqlite.-$$Lambda$PersistentModelVersion$3EIFYsUxnq4w1GB5yAimLeipvmc.accept(lambda)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$save$3$SQLiteStorageAdapter(SQLiteStorageAdapter.java:59)
        at com.amplifyframework.datastore.storage.sqlite.-$$Lambda$SQLiteStorageAdapter$JhGtXoPGeJ7lbNvRHxROfLs-WmE.run(lambda)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
        at java.lang.Thread.run(Thread.java:818)
     Caused by: com.amplifyframework.datastore.DataStoreException: Failure performing sync query to AppSync.
        at com.amplifyframework.datastore.appsync.AppSyncClient.lambda$sync$1(AppSyncClient.java:1)
        at com.amplifyframework.datastore.appsync.-$$Lambda$AppSyncClient$suvEma05s-p8v-66iE4cswLKYac.accept(lambda)
        at com.amplifyframework.api.aws.SingleArrayResultOperation$OkHttpCallback.onFailure(SingleArrayResultOperation.java:3)
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:510)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
        at java.lang.Thread.run(Thread.java:818) 
     Caused by: com.amplifyframework.api.ApiException: Could not retrieve the response body from the returned JSON
        at com.amplifyframework.api.aws.SingleArrayResultOperation$OkHttpCallback.onFailure(SingleArrayResultOperation.java:3) 
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:510) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
        at java.lang.Thread.run(Thread.java:818) 
     Caused by: javax.net.ssl.SSLHandshakeException: Connection closed by peer
        at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
        at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:318)
        at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.kt:367)
        at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.kt:325)
        at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:197)
    	at okhttp3.internal.connection.ExchangeFinder.findCo
04-11 15:51:28.060 6579-6612/ai.deepseed.sandbox W/amplify:aws-datastore: Error enqueuing mutation from subscription.
    com.amplifyframework.datastore.DataStoreException: Error during subscription.
        at com.amplifyframework.datastore.appsync.AppSyncClient.lambda$subscription$3(AppSyncClient.java:1)
        at com.amplifyframework.datastore.appsync.-$$Lambda$AppSyncClient$8gXTT9bgLZGP51blif6hkh7EwIc.accept(lambda)
        at com.amplifyframework.api.aws.SubscriptionEndpoint.requestSubscription(SubscriptionEndpoint.java:27)
        at com.amplifyframework.api.aws.SubscriptionOperation.lambda$start$0$SubscriptionOperation(SubscriptionOperation.java:4)
        at com.amplifyframework.api.aws.-$$Lambda$SubscriptionOperation$FXgdmzMG9ubNhC1f2NTwXyG53uo.run(lambda)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
        at java.lang.Thread.run(Thread.java:818)
     Caused by: com.amplifyframework.api.ApiException: Subscription timed out waiting for acknowledgement
        at com.amplifyframework.api.aws.SubscriptionEndpoint.requestSubscription(SubscriptionEndpoint.java:27) 
        at com.amplifyframework.api.aws.SubscriptionOperation.lambda$start$0$SubscriptionOperation(SubscriptionOperation.java:4) 
        at com.amplifyframework.api.aws.-$$Lambda$SubscriptionOperation$FXgdmzMG9ubNhC1f2NTwXyG53uo.run(lambda) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
        at java.lang.Thread.run(Thread.java:818) 
```

Thanks.

**Comment by atomicalsoftwares** [comment]
I should have probably noted: this is with a fresh install, I imagine it is happening once we're initialize the SDK.

**Comment by jpignata** [comment]
This should have been resolved in 1.1.1 -- let me know if that's not the case. Thank you!
